### PR TITLE
chore: fix windows CI

### PR DESCRIPTION
**Pull request by carllerche**
Trying to see if windows CI can be fixed.

### CODE REVIEW

File: tokio/tests/rt_common.rs [hunk 1]
```diff
@@ -601,6 +601,7 @@ rt_test! {
     }
 
     #[test]
+    #[cfg(not(windows))]
```
Comment (carllerche) [inline tokio/tests/rt_common.rs:604]: This test can't pass on windows. For now, I'm going to disable it. I will document why before completing the PR. I mostly want to see if CI passes.

File: tokio/src/net/tcp/split_owned.rs [hunk 2]
```diff
@@ -221,7 +221,7 @@ impl OwnedWriteHalf {
 impl Drop for OwnedWriteHalf {
     fn drop(&mut self) {
         if self.shutdown_on_drop {
-            let _ = self.inner.shutdown(Shutdown::Both);
+            let _ = self.inner.shutdown(Shutdown::Write);
```
Comment (carllerche) [inline tokio/src/net/tcp/split_owned.rs:224]: `Shutdown::Read` has no defined portable behavior. This is why it doesn't pass on windows.

**Comment by Matthias247** [comment]
I think this is now no longer has an equivalent behavior.

The idea behind shutting down the reading end was to emulate the same kind of behavior that e.g. socket APIs in Go and Java have: If you close the socket in those environments on one handle (e.g. a writing thread), it will unblock any operation blocked on receive. This helps if you have have one of those typical scenarios where there is a reader and a writer task and the reader is basically permanently blocked on the socket since it just reads in a loop. If it doesn't  get woken by the close it will be stuck there until the peer also closes.

Now doing this with shutdown instead of close is not exactly equivalent, but nevertheless was intended to do that job.

If that behavior is removed then any application which already relies on that behavior would get broken. The wakeup could probably be implemented on top of the native socket by grabbing the read waker from somewhere when the writer is closed, and storing and checking an extra "closed", but it's certainly more work.

This would now get balanced again breaking the API.

**Comment by carllerche** [comment]
`shutdown(Read)` has no *portable* behavior and the implementation as is did not work correctly across platforms.

Additionally, dropping a write half **should** only impact the write half of things, not the read. it is **incorrect** for Tokio to shutdown the read half when the write is dropped. This change is fixing a bug.

**Comment by carllerche** [comment]
@Darksonn IMO the API needs to be either fixed by making the change in this PR or deprecated w/ the reason that it cannot be implemented in a portable way and done again in 0.3. Which route do you want to take?

**Comment by Darksonn** [comment]
Considering that we are changing behavior on some OS with either option, I think it's ok to just fix it.

**Pull request merged**